app/src/main/java/com/gh4a/widget/CommitStatusBox.java

```diff
@@ -0,0 +1,211 @@
+            final int iconDrawableAttrId;
+            switch (state) {
+                case CommitStatus.STATE_ERROR:
+                case CommitStatus.STATE_FAILURE:
+                    iconDrawableAttrId = R.attr.commitStatusFailIcon;
+                    failingCount += 1;
+                    break;
+                case CommitStatus.STATE_SUCCESS:
+                    iconDrawableAttrId = R.attr.commitStatusOkIcon;
+                    successCount += 1;
+                    break;
+                default:
+                    iconDrawableAttrId = R.attr.commitStatusUnknownIcon;
+                    pendingCount += 1;
+                    break;
+            }
+            ImageView icon = statusRow.findViewById(R.id.iv_status_icon);
+            icon.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
+            icon.setImageResource(UiUtils.resolveDrawable(getContext(), iconDrawableAttrId));
+
+            TextView context = statusRow.findViewById(R.id.tv_context);
+            context.setText(status.getContext());
+
+            TextView description = statusRow.findViewById(R.id.tv_desc);
+            description.setText(status.getDescription());
+
+            mStatusContainer.addView(statusRow);
+        }
+
+        if (PullRequest.MERGEABLE_STATE_UNSTABLE.equals(mergableState) && pendingCount > 0) {
+            int resId = UiUtils.resolveDrawable(getContext(), R.attr.pullRequestMergeUnknownIcon);
+            mStatusIcon.setImageResource(resId);
+            mStatusLabel.setText(R.string.pull_merge_status_pending);
+        }
+
+        setSummaryText(failingCount, pendingCount, successCount);
+        setStatusesExpanded(failingCount + pendingCount > 0);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.row_commit_status:
+                CommitStatus status = (CommitStatus) v.getTag();
+                IntentUtils.launchBrowser(getContext(), Uri.parse(status.getTargetUrl()));
+                break;
+            case R.id.commit_status_header:
+                setStatusesExpanded(mStatusContainer.getVisibility() != View.VISIBLE);
+                break;
+        }
+    }
+
+    private void setSummaryText(int failingCount, int pendingCount, int successCount) {
+        String failingText = failingCount > 0
+                ? getContext().getString(R.string.check_failing, failingCount)
+                : "";
+        if (failingCount > 0 && pendingCount > 0) {
+            failingText += ", ";
+        }
+
+        String pendingText = pendingCount > 0
+                ? getContext().getString(R.string.check_pending, pendingCount)
+                : "";
+        if ((failingCount > 0 || pendingCount > 0) && successCount > 0) {
+            pendingText += ", ";
+        }
+
+        String successfulText = successCount > 0
+                ? getContext().getString(R.string.check_successful, successCount)
+                : "";
```

Comment:
This may be overkill, but this seems like it's suited for using an ArrayList + TextUtils.join().